compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -42,17 +42,17 @@ pub enum PassMode {
     ///
     /// The argument has a layout abi of `ScalarPair`.
     Pair(ArgAttributes, ArgAttributes),
-    /// Pass the argument after casting it, to either a single uniform or a
-    /// pair of registers. The bool indicates if a `Reg::i32()` dummy argument
-    /// is emitted before the real argument.
+    /// Pass the argument after casting it. See the `CastTarget` docs for details. The bool
+    /// indicates if a `Reg::i32()` dummy argument is emitted before the real argument.
     Cast(Box<CastTarget>, bool),
```

Comment:
Maybe move the bool to CastTarget field? That allows docs on the field itself and makes it more obvious when reading a use site.

Reply:
Hm, but then `CastTarget::llvm_type` would have to ignore the bool sine it's not part of the type.

IMO it would make more sense to turn this into a variant with named fields.

Reply:
Done.